core/src/main/java/org/hisp/dhis/android/core/fileresource/internal/FileResourceDownloadCall.kt

```diff
@@ -277,16 +280,17 @@ internal class FileResourceDownloadCall(
     private suspend fun <V> downloadFile(
         value: V,
         maxContentLength: Int?,
-        download: suspend (V) -> ByteArray?,
+        download: suspend (V, Long?) -> ByteArray?,
         fileResource: FileResource,
     ): FileResource? {
+        val contentLength = fileResource.contentLength()
         val acceptedContentLength = (maxContentLength == null) ||
-            (fileResource.contentLength() == null) ||
-            (fileResource.contentLength()!! <= maxContentLength)
+            (contentLength == null) ||
+            (min(fileResource.contentLength()!!.toLong(), DimensionSizeB.SMALL.maxSizeB) <= maxContentLength)
```

Comment:
could you avoid using !! changing `fileresource.contentLenght()` by `contentLenght`?